DNN Platform/Library/Entities/Tabs/TabController.cs

```diff
@@ -1469,6 +1469,33 @@ public TabCollection GetTabsByPortal(int portalId)
                                                             });
         }
 
+        /// <summary>
+        /// Get the actual visible tabs for a given portal id. 
+        /// System Tabs and Admin Tabs are excluded from the result set.
+        /// </summary>
+        /// <param name="portalId"></param>
+        /// <param name="adminTabId"></param>
+        /// <returns></returns>
+        public TabCollection GetTabsByPortal(int portalId, int adminTabId)
+        {
+            string cacheKey = string.Format(DataCache.TabCacheKey, portalId);

```

Comment:
This new method is going to introduce an undesired behavior.  We now have two methods, with different parameters using the same cache key to store data.  This could result in the expected/actual results being different.  A different key should be created based on the presence of an AdminTabId.

Also, given the possibility of large portals etc, is there any ability to do the filtering at the DB layer rather than with LINQ after?

Reply:
You're right. I should check if somewhere in the code we really need the full list of pages, including System and Admin pages. In this case, I would keep two methods with a different cache key.